tests/sedes/test_composite.py

```diff
@@ -0,0 +1,105 @@
+from eth_utils import (
+    decode_hex,
+    encode_hex,
+)
+import pytest
+
+import ssz
+from ssz.sedes import (
+    Container,
+    List,
+    Vector,
+    uint8,
+)
+
+
+@pytest.mark.parametrize(
+    ("value", "serialized"),
+    (
+        ((), "0x00000000"),
+        ((0xaa,), "0x01000000aa"),
+        ((0xaa, 0xbb, 0xcc), "0x03000000aabbcc"),
+        ((0xaa,) * 256, "0x00010000" + "aa" * 256),
+        ((0xaa,) * (256**2 - 1), "0xffff0000" + "aa" * (256**2 - 1)),
+    ),
+)
+def test_list(value, serialized):
+    sedes = List(uint8)
+    assert encode_hex(ssz.encode(value, sedes)) == serialized
+    assert ssz.decode(decode_hex(serialized), sedes) == value
+
+
+@pytest.mark.parametrize(
+    ("value", "serialized"),
+    (
+        ((), "0x"),
+        ((0xaa,), "0xaa"),
+        ((0xaa, 0xbb, 0xcc), "0xaabbcc"),
+    ),
+)
+def test_tuple_of_static_sized_entries(value, serialized):
+    sedes = Vector(len(value), uint8)
+    assert encode_hex(ssz.encode(value, sedes)) == serialized
+    assert ssz.decode(decode_hex(serialized), sedes) == value
+
+
+@pytest.mark.parametrize(
+    ("value", "serialized"),
+    (
+        (((),), "0x0400000000000000"),
+        (((0xaa,),), "0x0500000001000000aa"),
+        (((0xaa, 0xbb, 0xcc),), "0x0700000003000000aabbcc"),
+        (((), (), ()), "0x0c000000000000000000000000000000"),
+        (
+            ((0xaa,), (0xbb, 0xcc), (0xdd, 0xee, 0xff)),
+            "0x1200000001000000aa02000000bbcc03000000ddeeff"
+        ),
+    )
+)
+def test_tuple_of_dynamic_sized_entries(value, serialized):
+    sedes = Vector(len(value), List(uint8))
+    assert encode_hex(ssz.encode(value, sedes)) == serialized
+    assert ssz.decode(decode_hex(serialized), sedes) == value
+
+
+@pytest.mark.parametrize(
+    ("value", "serialized"),
+    (
+        # ((), "0x"),
```

Comment:
It seems related to https://github.com/ethereum/eth2.0-specs/issues/854. Maybe add a TODO comment in this PR and we can come back later.

Reply:
whoops, that actually works, not sure why it's commented out. Enabled it again.